grid_map_geo/src/terrain_loader.cpp

```diff
@@ -0,0 +1,150 @@
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+/**
+ * @brief Node to test planner in the view utiltiy map
+ *
+ *
+ * @author Jaeyoung Lim <jalim@ethz.ch>
+ */
+
+#include <grid_map_geo/grid_map_geo.h>
+#include <grid_map_msgs/GridMap.h>
+#include <ros/ros.h>
+#include <grid_map_ros/GridMapRosConverter.hpp>
+
+#include <gdal/ogr_spatialref.h>
+
+constexpr int ESPG_WGS84 = 4326;
+constexpr int ESPG_CH1903_LV03 = 21781;
+
+void transformCoordinates(int src_coord, const double &x, const double &y, const double &z, int tgt_coord, double &x_t,
+                          double &y_t, double &z_t) {
+  OGRSpatialReference source, target;
+  source.importFromEPSG(src_coord);
+  target.importFromEPSG(tgt_coord);
+
+  OGRPoint p;
+  p.setX(x);
+  p.setY(y);
+  p.setZ(z);
+  p.assignSpatialReference(&source);
+
+  p.transformTo(&target);
+  x_t = p.getX();
+  y_t = p.getY();
+  z_t = p.getZ();
+  return;
+}
+
+void LoadTerrainFromVrt(std::string path, const Eigen::Vector3d &query_position, const Eigen::Vector2d &offset,
+                        grid_map::GridMap &map, grid_map::GridMap &vrt_map_object) {
+  std::shared_ptr<GridMapGeo> vrt_map = std::make_shared<GridMapGeo>();
+
+  Eigen::Vector3d query_position_lv03 = query_position;
+  /// Convert LV03 to WGS84
+  Eigen::Vector3d map_center_wgs84;  // Map center in WGS84
+  transformCoordinates(ESPG_CH1903_LV03, query_position_lv03(0), query_position_lv03(1), query_position_lv03(2),
+                       ESPG_WGS84, map_center_wgs84.x(), map_center_wgs84.y(), map_center_wgs84.z());
+
+  std::cout << "Loading VRT Map:" << std::endl;
+  std::cout << "  - map_center_wgs84:" << map_center_wgs84.transpose() << std::endl;
+  /// TODO: Discover extent from corners
+  Eigen::Vector2d extent_wgs84(0.5, 0.5);
+  vrt_map->initializeFromVrt(path, map_center_wgs84.head(2), extent_wgs84);
+  std::cout << "  - Success!" << std::endl;
+
+  /// TODO: Loaded VRT map
+  std::cout << "Reprojecting map" << std::endl;
+  Eigen::Vector2d extent_lv03(12000.0, 12000.0);
+  double resolution{100.0};
+  Eigen::Vector2d position{Eigen::Vector2d::Zero()};
+  map.setGeometry(extent_lv03, resolution, position);
+  map.add("elevation");
+  map.setFrameId("map");
+  for (grid_map::GridMapIterator iterator(map); !iterator.isPastEnd(); ++iterator) {
+    const grid_map::Index index = *iterator;
+    Eigen::Vector2d cell_position;  // Position of cell relative to map coordinates
+    map.getPosition(index, cell_position);
+    auto cell_position_lv03 = cell_position + query_position_lv03.head(2);  // Position of cell in CH1903/LV03
+    double dummy;
+    Eigen::Vector2d cell_position_wgs84;
+    transformCoordinates(ESPG_CH1903_LV03, cell_position_lv03(0), cell_position_lv03(1), cell_position_lv03(2),
```

Comment:
Rather than transform individual coordinates, do you think doing a warp operation in `gdal` would lead to higher performance?

Reply:
@Ryanf55 Definitely, but not sure how we can do this within the cpp API